app/src/main/java/com/github/se/project/ui/components/DisplayLessonDetails.kt

```diff
@@ -38,18 +37,36 @@ fun DisplayLessonDetails(lesson: Lesson, studentProfile: Profile, modifier: Modi
       colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant)) {
         Column(
             modifier = Modifier.padding(16.dp), verticalArrangement = Arrangement.spacedBy(16.dp)) {
+
               // Student information section
               StudentInfoSection(studentProfile)
 
               Divider(color = MaterialTheme.colorScheme.outlineVariant)
 
+              // Time and date section
+              Row(
+                  modifier = Modifier.fillMaxWidth(),
+                  horizontalArrangement = Arrangement.spacedBy(16.dp),
+                  verticalAlignment = Alignment.CenterVertically) {
+                    Row(
+                        horizontalArrangement = Arrangement.spacedBy(8.dp),
+                        verticalAlignment = Alignment.CenterVertically,
+                        modifier = Modifier.weight(1f)) {
+                          Icon(
+                              imageVector = Icons.Default.DateRange,
+                              contentDescription = null,
+                              tint = MaterialTheme.colorScheme.primary)
+                          Text(
+                              text = formatDate(lesson.timeSlot),
+                              style = MaterialTheme.typography.bodyMedium,
+                              modifier = Modifier.testTag("lessonTime"))
+                        }
+                  }
```

Comment:
I suppose the composable doesn't need to exist if used only once, but having it separate could have been useful for future screens